packages/playwright/src/AxeCoreWrapper/DefaultAxeCoreWrapper.cs

```diff
@@ -1,108 +1,297 @@
+        }
+
+        private static List<AxeRuleMetadata> DeserializeRuleMetadata(object jsonObject)
+        {
+            string result = JsonConvert.SerializeObject(jsonObject);
+            var results = JsonConvert.DeserializeObject<List<AxeRuleMetadata>>(result);
+
+            if (results is null)
+            {
+                throw new Exception();
+            }
+
+            return results;
+        }
+
+        /// <summary>
+        /// Runs axe via legacyScan.js at a specific context, which will be passed as-is to Selenium for legacyScan.js to interpret, and
+        /// parses/handles the legacyScan.js output per the current builder options.
+        /// </summary>
+        /// <param name="rawContextArg">The value to pass as-is to legacyScan.js to use as the axe.run "context" argument</param>
+        private async Task<AxeResult> RunInner(IFrame frame, object rawContextArg, AxeRunOptions? runOptions)
+        {
+            await ConfigureAxe(frame);
+
+
+            if (runOptions == null)
+            {
+                runOptions = new AxeRunOptions
+                {
+                    Iframes = true
+                };
+            }
+
+            var runPartialExists = await RunPartialExists(frame);
+
+            if (!runPartialExists)
+            {
+                await m_axeContentEmbedder.EmbedAxeCoreIntoPage(frame.Page, runOptions.Iframes, true);
+                return await EvaluateAxeRun(frame.Page, null, runOptions);
+            }
+            else
+            {
+                return await AnalyzeAxeRunPartial(frame, rawContextArg, runOptions);
+            }
+
+
+        }
+
+        private async Task<AxeResult> AnalyzeAxeRunPartial(IFrame frame, object rawContextArg, AxeRunOptions runOptions)
+        {
+            string rawOptionsArg = JsonConvert.SerializeObject(runOptions);
+            var partialResults = await RunPartialRecursive(frame, rawOptionsArg, rawContextArg, true, !runOptions.Iframes.HasValue || runOptions.Iframes.Value);
+            return await IsolatedFinishRun(frame, partialResults.ToArray(), rawOptionsArg);
+        }
+
+        private async Task<List<object?>> RunPartialRecursive(
+                                                        IFrame frame,
+                                                        object options,
+                                                        object context,
+                                                        bool isTopLevel,
+                                                        bool iframes
+                                                        )
+        {
+            if (!isTopLevel)
+            {
+                await ConfigureAxe(frame);
+            }
+
+            var partialResults = new List<object?>();
+
+            try
+            {
+                string partialRes = await frame.EvaluateAsync<string>(EmbeddedResourceProvider.ReadEmbeddedFile("runPartial.js"), new[] { context, options });

```

Comment:
[The equivalent bit of `@axe-core/playwright`](https://github.com/dequelabs/axe-core-npm/blob/2bdd8a8fce420abb1ec78adea00690f3a83a898f/packages/playwright/src/index.ts#L218) uses a different order of execution here with the suggestion that it "avoids any potential problems with the DOM becoming out of sync".

That is, for a case with 3 frames (A contains B contains C):

Order of execution with this code:
1. await runPartial(A)
2. await getFrameContexts(A)
3. await shadowSelect(B in A)
4. await runPartial(B)
5. await getFrameContexts(B)
6. await shadowSelect(C in B)
7. await runPartial(C)
8. await getFrameContexts(C)
9. await newPage, await finishRun

Order of execution with `@axe-core/playwright`:
1. await getFrameContexts(A)
2. start (but do not await) runPartial(A)
3. await shadowSelect(B in A)
4. await getFrameContexts(B)
5. start (but do not await) runPartial(B)
6. await shadowSelect(C in B)
7. await getFrameContexts(C)
8. start (but do not await) runPartial(C)
9. await previously-started runPartial(A)
10. await previously-started runPartial(B)
11. await previously-started runPartial(C)
12. await newPage, await finishRun

Should this code be operating closer to how `@axe-core/playwright` is set up? I think it would be preferable for performance reasons to be able to parallelize the frames' `runPartial` calls, but the comment in `@axe-core/playwright` suggests that it might be a correctness issue, not just a performance once.